src/Integration.Vsix/SonarAnalyzerCollisionManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="PackageCommandManager.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServices;
+using SonarLint.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class SonarAnalyzerCollisionManager
+    {
+        private readonly Workspace workspace;
+
+        private static readonly AssemblyName AnalyzerAssemblyName =
+            new AssemblyName(typeof(SonarAnalysisContext).Assembly.FullName);
+        internal /*for testing purposes*/ static readonly Version AnalyzerVersion = AnalyzerAssemblyName.Version;
+        internal /*for testing purposes*/ static readonly string AnalyzerName = AnalyzerAssemblyName.Name;
+
+        public SonarAnalyzerCollisionManager(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            var componentModel = (IComponentModel)serviceProvider.GetService(typeof(SComponentModel));
+            workspace = componentModel.GetService<VisualStudioWorkspace>();
+
+            SonarAnalysisContext.ShouldAnalysisBeDisabled =
+                tree => ShouldAnalysisBeDisabledOnTree(tree);
+        }
+
+        private bool ShouldAnalysisBeDisabledOnTree(SyntaxTree tree)
+        {
+            if (tree == null)
+            {
+                return false;
+            }
+
+            IEnumerable<AnalyzerReference> references = workspace?.CurrentSolution?.GetDocument(tree)?.Project?.AnalyzerReferences;
+            return HasConflictingAnalyzerReference(references);
+        }
+
+        internal /*for testing purposes*/ static bool HasConflictingAnalyzerReference(IEnumerable<AnalyzerReference> references)
+        {
+            if (references == null)
+            {
+                return false;
+            }
+
+            foreach (var reference in references.Where(a => a.Display == AnalyzerName))
+            {
+                Version version = (reference.Id as AssemblyIdentity)?.Version;
+                return version != AnalyzerVersion;
```

Comment:
@mjcheetham As for having different versions: The `AnalyzerReferences` are (usually) added from NuGet, so we don't really expect to find multiple versions of SonarAnalyzer in this list. However, you could specify manually multiple different versions of an analyzer in the csproj. I'm unsure if VS will load them, or fail to handle properly same-named analyzers. If we consider that VS can load them and we have SonarAnalyzers version A and B added. VS will report issues multiple times on the same location. If we add SonarLint to the picture with SonarAnalyzer version B, and we are doing the processing in A, B order, then you are right we'll find A != B and thus we disable B which is a different experience then what we should have. 

I'll change the check accordingly, although I don't think we'd have faced this issue. (Adding a test for it too)